### PR TITLE
fix npm publish .DS_Store

### LINKED ISSUE #10112: .DS_Store files shipped in npm package
### Version
2.6.10

### Reproduction link
[https://unpkg.com/vue@2.6.10/src/.DS_Store](https://unpkg.com/vue@2.6.10/src/.DS_Store)

### Steps to reproduce
```
wget https://unpkg.com/vue@2.6.10/src/.DS_Store
wget https://unpkg.com/vue@2.6.10/src/compiler/.DS_Store
wget https://unpkg.com/vue@2.6.10/src/platforms/weex/.DS_Store
wget https://unpkg.com/vue@2.6.10/src/platforms/weex/compiler/.DS_Store
wget https://unpkg.com/vue@2.6.10/src/platforms/.DS_Store
```
### What is expected?
```
--2019-06-06 18:17:31--  https://unpkg.com/vue@2.6.10/src/.DS_Store
Resolving unpkg.com (unpkg.com)... 104.16.124.175, 104.16.123.175, 104.16.122.175, ...
Connecting to unpkg.com (unpkg.com)|104.16.124.175|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2019-06-06 18:17:31 ERROR 404: Not Found.

--2019-06-06 18:17:31--  https://unpkg.com/vue@2.6.10/src/compiler/.DS_Store
Resolving unpkg.com (unpkg.com)... 104.16.124.175, 104.16.123.175, 104.16.122.175, ...
Connecting to unpkg.com (unpkg.com)|104.16.124.175|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2019-06-06 18:17:32 ERROR 404: Not Found.

--2019-06-06 18:17:32--  https://unpkg.com/vue@2.6.10/src/platforms/weex/.DS_Store
Resolving unpkg.com (unpkg.com)... 104.16.124.175, 104.16.123.175, 104.16.122.175, ...
Connecting to unpkg.com (unpkg.com)|104.16.124.175|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2019-06-06 18:17:33 ERROR 404: Not Found.

--2019-06-06 18:17:33--  https://unpkg.com/vue@2.6.10/src/platforms/weex/compiler/.DS_Store
Resolving unpkg.com (unpkg.com)... 104.16.124.175, 104.16.123.175, 104.16.122.175, ...
Connecting to unpkg.com (unpkg.com)|104.16.124.175|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2019-06-06 18:17:33 ERROR 404: Not Found.

--2019-06-06 18:17:33--  https://unpkg.com/vue@2.6.10/src/platforms/.DS_Store
Resolving unpkg.com (unpkg.com)... 104.16.124.175, 104.16.123.175, 104.16.122.175, ...
Connecting to unpkg.com (unpkg.com)|104.16.124.175|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2019-06-06 18:17:34 ERROR 404: Not Found.
```
### What is actually happening?
```
--2019-06-06 18:15:56--  https://unpkg.com/vue@2.6.10/src/.DS_Store
Resolving unpkg.com (unpkg.com)... 104.16.124.175, 104.16.123.175, 104.16.122.175, ...
Connecting to unpkg.com (unpkg.com)|104.16.124.175|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6148 (6.0K) [null]
Saving to: ‘.DS_Store’

.DS_Store                                             100%[=======================================================================================================================>]   6.00K  --.-KB/s    in 0s      

2019-06-06 18:15:57 (69.3 MB/s) - ‘.DS_Store’ saved [6148/6148]

--2019-06-06 18:15:57--  https://unpkg.com/vue@2.6.10/src/compiler/.DS_Store
Resolving unpkg.com (unpkg.com)... 104.16.124.175, 104.16.123.175, 104.16.122.175, ...
Connecting to unpkg.com (unpkg.com)|104.16.124.175|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6148 (6.0K) [null]
Saving to: ‘.DS_Store.1’

.DS_Store.1                                           100%[=======================================================================================================================>]   6.00K  --.-KB/s    in 0s      

2019-06-06 18:15:57 (65.4 MB/s) - ‘.DS_Store.1’ saved [6148/6148]

--2019-06-06 18:15:57--  https://unpkg.com/vue@2.6.10/src/platforms/weex/.DS_Store
Resolving unpkg.com (unpkg.com)... 104.16.124.175, 104.16.123.175, 104.16.122.175, ...
Connecting to unpkg.com (unpkg.com)|104.16.124.175|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6148 (6.0K) [null]
Saving to: ‘.DS_Store.2’

.DS_Store.2                                           100%[=======================================================================================================================>]   6.00K  --.-KB/s    in 0s      

2019-06-06 18:15:57 (75.0 MB/s) - ‘.DS_Store.2’ saved [6148/6148]

--2019-06-06 18:15:57--  https://unpkg.com/vue@2.6.10/src/platforms/weex/compiler/.DS_Store
Resolving unpkg.com (unpkg.com)... 104.16.124.175, 104.16.123.175, 104.16.122.175, ...
Connecting to unpkg.com (unpkg.com)|104.16.124.175|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6148 (6.0K) [null]
Saving to: ‘.DS_Store.3’

.DS_Store.3                                           100%[=======================================================================================================================>]   6.00K  --.-KB/s    in 0s      

2019-06-06 18:15:57 (33.9 MB/s) - ‘.DS_Store.3’ saved [6148/6148]

--2019-06-06 18:15:57--  https://unpkg.com/vue@2.6.10/src/platforms/.DS_Store
Resolving unpkg.com (unpkg.com)... 104.16.124.175, 104.16.123.175, 104.16.122.175, ...
Connecting to unpkg.com (unpkg.com)|104.16.124.175|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6148 (6.0K) [null]
Saving to: ‘.DS_Store.4’

.DS_Store.4                                           100%[=======================================================================================================================>]   6.00K  --.-KB/s    in 0s      

2019-06-06 18:15:57 (38.7 MB/s) - ‘.DS_Store.4’ saved [6148/6148]
```

### PR DESCRIPTION
**Pull request by theskillwithin**
**What kind of change does this PR introduce?** (check at least one)
- [x ] Bugfix
**Does this PR introduce a breaking change?** (check one)
- [ ] Yes
- [x ] No

prevent .DS_Store from being publish

see: https://unpkg.com/browse/vue@2.6.12/src/

Close #10112

### CODE REVIEW

**Comment by theskillwithin** [comment]
https://github.com/vuejs/vue/issues/10112

File: package.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
   "files": [
     "src",
```
Comment (posva) [inline package.json:11]: In order for the ignore rule to work, this one has to be more precise (like the other rules with extensions)

**Review by posva — CHANGES_REQUESTED** [review]
Review based on testing with https://unpkg.com/browse/@posva/test-ds-store@1.0.3/

**Comment by theskillwithin** [comment]
@posva I think someone should make a new PR, I deleted this forked repo so seems I can't edit this PR anymore

**Comment by posva** [comment]
Thanks for the initiative though!

**Pull request closed without merging**